src/client/routes/user_reports.tsx

```diff
@@ -1,14 +1,27 @@
 import { DateToYMDString } from "../helpers/datetime";
-import { useState } from "react";
+import { getRouteApi } from "@tanstack/react-router";
 import UserReportFrame from "../components/user_report_frame";
 
+const routeApi = getRouteApi("/user_reports");
+
 export default function UserReports() {
+  const searchParams = routeApi.useSearch();
+  const navigate = routeApi.useNavigate();
+
+  const navigateToDate = (newDateString: string) => {
+    navigate({ search: Object.assign({}, searchParams, { from: newDateString }) });
```

Comment:
I think one can write this as something like `{...searchParams, from: newDateString}`?

Reply:
One day, I will get used to the spread syntax... 🙈 Fixed.